packages/api/src/controllers/user.ts

```diff
@@ -1525,28 +1528,62 @@ app.post(
       { limit },
     );
 
-    const results = [];
+    const results = {
+      migrated: [],
+      flagged: [],
+    };
+
+    const OLD_USER_CUTOFF = Date.now() - ms("12 months");
 
     for (const user of users) {
+      if (isFakeEmail(user.email) && !user.admin) {
+        if (actuallyMigrate) {
+          await db.user.update(user.id, { toBeDeleted: true });
+        }
+        results.flagged.push({ id: user.id, email: user.email });
+        continue;
+      }
+
+      const isOldUser = user.createdAt < OLD_USER_CUTOFF && !user.admin;
+      const isInactive =
+        user.lastSeen < OLD_USER_CUTOFF &&
+        user.lastStreamedAt < OLD_USER_CUTOFF;
+
+      if (isOldUser && isInactive) {
+        const [tokens] = await db.apiToken.find({ userId: user.id });
+        const allTokensOld =
+          tokens.length === 0 ||
+          tokens.every((token) => token.lastSeen < OLD_USER_CUTOFF);
+
+        if (allTokensOld) {
```

Comment:
**nit:** I think this whole logic could be made clearer by having a single condition like 
```
shouldDeleteUser = !user.admin && (isFakeEmail(user.email) || (isOldUser && isInactive && allTokensOld))
```
WDYT?

Reply:
Followed your suggestion! should be much clearer now